src/v/redpanda/admin/api-doc/debug.json

```diff
@@ -395,6 +395,165 @@
               ]
             }
           ]
+        },
+        {
+            "path": "/v1/debug/bundle/actions/create",
+            "operations": [
+                {
+                    "method": "POST",
+                    "summary": "Start creating a debug bundle",
+                    "type": "void",
+                    "nickname": "start_debug_bundle",
+                    "produces": [
+                        "application/json"
+                    ],
+                    "parameters": [
+                        {
+                            "name": "logs-since",
+                            "in": "query",
+                            "required": false,
+                            "type": "string"
+                        },
+                        {
+                            "name": "logs-until",
+                            "in": "query",
+                            "required": false,
+                            "type": "string"
+                        },
+                        {
+                            "name": "logs-size-limit",
+                            "in": "query",
+                            "required": false,
+                            "type": "string"
+                        },
+                        {
+                            "name": "metrics-interval",
+                            "in": "query",
+                            "required": false,
+                            "type": "string"
+                        }
+                    ],
+                    "responses": {
+                        "202": {
+                            "description": "Successfully started debug bundle"
+                        },
+                        "429": {
+                            "description": "A bundle is already running"
+                        },
+                        "500": {
+                            "description": "Internal Server error"
+                        }
+                    }
+                }
+            ]
+        },
+        {
+            "path": "/v1/debug/bundle/files/{filename}",
```

Comment:
Should the URL accept a filename? Since we only support one bundle at a time, we can simplify the URLs by dropping the filename parameter, and then generalize the URLs when we later support multiple files? Even then, I would assume that we'd want to generate IDs of some sort rather than actual couple the implementation detail of a filename to the API.

wdyt?

Reply:
> Should the URL accept a filename? Since we only support one bundle at a time, we can simplify the URLs by dropping the filename parameter, and then generalize the URLs when we later support multiple files?

My opinion is that we should keep the API path as is to maintain compatibility between the MVP and full erversion

> Even then, I would assume that we'd want to generate IDs of some sort rather than actual couple the implementation detail of a filename to the API.

This makes sense.  Should be some sort of maybe date encoded string that signifies when the bundle was created.

Reply:
> > Should the URL accept a filename? Since we only support one bundle at a time, we can simplify the URLs by dropping the filename parameter, and then generalize the URLs when we later support multiple files?
> 
> My opinion is that we should keep the API path as is to maintain compatibility between the MVP and full erversion

Coding so other teams have less work in the future makes sense to me as well. Otherwise, the RPK team would have to make changes for the next version of this feature.

> 
> > Even then, I would assume that we'd want to generate IDs of some sort rather than actual couple the implementation detail of a filename to the API.
> 
> This makes sense. Should be some sort of maybe date encoded string that signifies when the bundle was created.

To make sure I understand this point, the suggestion is to:
- When someone issues `/v1/debug/bundle/actions/create`, return an ID that maps to the debug bundle on disk
-  Change the path for other URIs to from `/v1/debug/bundle/files/{filename}` -> `/v1/debug/bundle/files/{file ID}`

Reply:
> This makes sense. Should be some sort of maybe date encoded string that signifies when the bundle was created.

The way this would classically be done with REST is to define

   `GET /v1/debug/bundle`

to return a list of bundles and then have another URL

   `GET /v1/debug/bundle/{id}` for fetching the target bundle.

The list of bundles can have whatever metadata we want, along with an id.

The id can literally be the filename, it's just that the API can be designed in a way to not force that.

Reply:
Maybe I am missing something, but this description is the same as what is proposed in the PRD

> > This makes sense. Should be some sort of maybe date encoded string that signifies when the bundle was created.
> 
> The way this would classically be done with REST is to define
> 
> `GET /v1/debug/bundle`
> 
> to return a list of bundles and then have another URL

From the PRD:
`GET /v1/debug/bundle/files` returns the list of bundles available on disk. This endpoint is **not** going in the MVP though.

> `GET /v1/debug/bundle/{id}` for fetching the target bundle.

From the PRD: `GET /v1/debug/bundle/files/{filename}` fetches the target bundle. This endpoint definitely going in the MVP.

--

To get back to the original question in this thread: "Should the filename be in the uri path?"

I do not see a problem with keeping it in the path since it minimizes the number of changes to do in the future. The only reason I can see for removing the filename is **if** the time between this release and the future release is large (maybe 2 quarters? I don't know what a reasonable period of time here is). In this scenario, users would be forced to put the filename of a known bundle in the path and I can see that being a poor UX (it is a little annoying).

Are there any other reasons to remove the filename @dotnwat ?

Reply:
> The way this would classically be done with REST is to define
> 
> `GET /v1/debug/bundle`
> 
> to return a list of bundles and then have another URL

PRD does specify `GET /v1/debug/bundle/files` would return an list of files.  Can probably update that to return a list of objects containing the file and metadata.

> `GET /v1/debug/bundle/{id}` for fetching the target bundle.
> 
> The list of bundles can have whatever metadata we want, along with an id.
>
> The id can literally be the filename, it's just that the API can be designed in a way to not force that.

Yeah the 'id' is whatever we want (the path in the PRD is `/v1/debug/budle/files/{id}`.  I'd say if we want to include any additional metadata into the PRD, we should make those recommendations now.  I like the idea that `GET /v1/debug/bundle/files` returns more info rather than just a list of files.

As I've said previously, I don't want to stray too far from the PRD if at all possible.

Reply:
Oh also since the MVP right now isn't implementing `/v1/debug/bundle/files` we have time to further refine this.